src/material/badge/_badge-theme.scss

```diff
@@ -102,6 +102,14 @@ $_badge-structure-emitted: false !default;
     position: relative;
   }
 
+  // The badge should make sure its host is overflow visible so that the badge content
+  // can be rendered outside of the element. Some components such as <mat-icon> explicitly
+  // style `overflow: hidden` so this requires extra specificity so that it does not
+  // depend on style load order.
+  .mat-badge.mat-badge {
+    overflow: visible;
```

Comment:
Optional: should we just make this `!important`?

Reply:
If so let's do it in another PR just because we want to get this in before the release goes out so the "next" version doesn't have broken icon badges, and I've got a TGP for this version of the fix. Technically this seems to be a good candidate for actually using important than its critical to the component working